### DISCUSSION TITLE
Camera clipping with near and far

### DISCUSSION THREAD

**Discussion by Daoudou** [openscenegraph/OpenSceneGraph#1284] [Q&A] [answered]
Hello

I have a small problem (for a change) I have an OSG scene with a view matrix and a projection matrix that I get from an Opengl scene,
like this :

` 
        osg::Matrixd modelViewMatrix(m_dModelMatrix[0], m_dModelMatrix[1], m_dModelMatrix[2], m_dModelMatrix[3],
            m_dModelMatrix[4], m_dModelMatrix[5], m_dModelMatrix[6], m_dModelMatrix[7],
            m_dModelMatrix[8], m_dModelMatrix[9], m_dModelMatrix[10], m_dModelMatrix[11],
            m_dModelMatrix[12], m_dModelMatrix[13], m_dModelMatrix[14], m_dModelMatrix[15]
        );
 
        osg::Matrixd projectionMatrix(m_dProjetMatrix[0], m_dProjetMatrix[1], m_dProjetMatrix[2], m_dProjetMatrix[3],
            m_dProjetMatrix[4], m_dProjetMatrix[5], m_dProjetMatrix[6], m_dProjetMatrix[7],
            m_dProjetMatrix[8], m_dProjetMatrix[9], m_dProjetMatrix[10], m_dProjetMatrix[11],
            m_dProjetMatrix[12], m_dProjetMatrix[13], m_dProjetMatrix[14], m_dProjetMatrix[15]
        );
 
        Camera->setViewMatrix(modelViewMatrix);
        Camera->setProjectionMatrix(projectionMatrix);`

And my problem comes from the fact that I have clipping on my OSG scene that I don't have on my Opengl scene:
I specify: (For the 2 scenes)
- The near is -10000
- The far is 10000

![Capture1](https://github.com/openscenegraph/OpenSceneGraph/assets/98310378/9f0ab563-3a1a-4509-b2a7-daefcab3cc59)

But if I use the setting for the camera:

`Camera->setComputeNearFarMode(osg::CullSettings::DO_NOT_COMPUTE_NEAR_FAR);`

I no longer have clipping. This could solve the problem, however certain functions require near and far otherwise I get a black screen.

![Capture2](https://github.com/openscenegraph/OpenSceneGraph/assets/98310378/b94fabd5-aa88-4a05-bf56-6671a77cac42)

My question is therefore: How could I apply the view of the second image without deactivating the near and far calculation?

Thanks in advance

Daoudou

**Comment by robertosfield** [comment, marked as answer]
The pnear and far planes, set by the projection matrix, are always part of OSG/OpenGL rendering, the DO_NOT_COMPUTE_NEAR_FAR setting only changes how the OSG sets up the near and far values, adjusting the projection matrix so that the near and far planes encompass the extents of the scene that passes the view frustum test.

If switching off the DO_NOT_COMPUTE_NEAR_FAR helps you see more of the scene then it's because the near and far planes (set via the projection matrix) are too far away for the extents of your scene, so you need to set the default values better for your scene.
